addon/helpers/optional.js

```diff
@@ -1,8 +1,8 @@
 import { helper } from 'ember-helper';
 
-export function optional([action]) {
+export function optional([action, ...args]) {
   if (typeof action === 'function') {
-    return action;
+    return action.bind(this, ...args);
```

Comment:
Please avoid binding `this`, because that will screw up the context of (closure) actions.